{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c142f605-dd77-48d0-b82b-f563cf85c1f7",
   "metadata": {},
   "source": [
    "### Topic 1. Not all VEP-representative isoforms are in the MSKCC and UniProt isoform list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f6ead22-1aac-41f4-88c5-0b9e6b43e7ec",
   "metadata": {},
   "source": [
    "<font color=\"red\">Terminology alert!</font>\n",
    "- MSK isoforms: https://github.com/mskcc/vcf2maf/blob/main/data/isoform_overrides_at_mskcc\n",
    "- UniProt isoforms: https://github.com/mskcc/vcf2maf/blob/main/data/isoform_overrides_uniprot\n",
    "\n",
    "Among many isoforms that encompasses a variant, VEP selects a representative transcript writes only its information in its 'Transcript_ID', 'HGVSc', 'HGVSp', etc. columns\n",
    "It's said,"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22619754-7a43-48c2-abb8-2b8c9abca25a",
   "metadata": {},
   "source": [
    "<i>\"To identify this transcript, we consider, where available, evidence of functional potential (such as evolutionary conservation of a coding region, transcript expression levels), transcript length and evidence from other resources (such as concordance with the APPRIS1 ’principal isoform’ and with the UniProt/Swiss-Prot ‘canonical isoform - functional, expressed, conserved, has consensus; if not choose longest’).\"</i> [link: http://mart.ensembl.org/info/genome/genebuild/canonical.html]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b459482-d0d0-4cd3-a8be-02bf21a6f144",
   "metadata": {},
   "source": [
    "Since I was told that running `MafAnnotator.py` using its protein querying option emits errors, probably due to the input variants not labeled with isoforms matching that of OncoKB, I tried to find the proportion of isoforms matching that of OncoKB (isoforms from either 'MSK isoforms' or 'UniProt isoforms'). Since I knew VEP maf files also included an 'all_effects' column that includes information for all other VEP isoforms, I might probably be able to re-match these isoforms to that of OncoKB."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9337f97f-833f-4d25-8976-2d11f122fa92",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "68438f54-3fb7-49de-9170-cf62bec7783a",
   "metadata": {},
   "outputs": [],
   "source": [
    "project = \"18328\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "7a1d7ef7-4eb3-4657-af1b-1178c4e13f26",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_29513/2203655544.py:1: DtypeWarning: Columns (0,5,6,113,114,145,156) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  flt = pd.read_table(f\"{project}/results/cohort_filtered.maf\") # here, the project directory has been soft-linked to this directory\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genomic_vcf37_source</th>\n",
       "      <th>chr_hg19</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>ref_hg19</th>\n",
       "      <th>alt_hg19</th>\n",
       "      <th>clinical_significance_brcaexchange</th>\n",
       "      <th>brcaexchange</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>...</th>\n",
       "      <th>HIGHEST_LEVEL</th>\n",
       "      <th>CITATIONS</th>\n",
       "      <th>LEVEL_Dx1</th>\n",
       "      <th>LEVEL_Dx2</th>\n",
       "      <th>LEVEL_Dx3</th>\n",
       "      <th>HIGHEST_DX_LEVEL</th>\n",
       "      <th>LEVEL_Px1</th>\n",
       "      <th>LEVEL_Px2</th>\n",
       "      <th>LEVEL_Px3</th>\n",
       "      <th>HIGHEST_PX_LEVEL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7577094</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21445056</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCL</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7578190</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>56477631</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERBB3</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7574003</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>19336573;27759562;16007150;11753428;11900253;2...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCL</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>115256530</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>LEVEL_3A</td>\n",
       "      <td>10574788;9219684;20194776;23515407;26619011;29...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 158 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  genomic_vcf37_source chr_hg19   pos_hg19 ref_hg19 alt_hg19  \\\n",
       "0                  NaN       17    7577094        G        A   \n",
       "1                  NaN       17    7578190        T        C   \n",
       "2                  NaN       12   56477631        T        A   \n",
       "3                  NaN       17    7574003        G        A   \n",
       "4                  NaN        1  115256530        G        T   \n",
       "\n",
       "  clinical_significance_brcaexchange brcaexchange Hugo_Symbol  Entrez_Gene_Id  \\\n",
       "0                                NaN          NaN        TP53               0   \n",
       "1                                NaN          NaN        TP53               0   \n",
       "2                                NaN          NaN       ERBB3               0   \n",
       "3                                NaN          NaN        TP53               0   \n",
       "4                                NaN          NaN        NRAS               0   \n",
       "\n",
       "  Center  ... HIGHEST_LEVEL  \\\n",
       "0      .  ...           NaN   \n",
       "1      .  ...           NaN   \n",
       "2      .  ...           NaN   \n",
       "3      .  ...           NaN   \n",
       "4      .  ...      LEVEL_3A   \n",
       "\n",
       "                                           CITATIONS LEVEL_Dx1 LEVEL_Dx2  \\\n",
       "0                                           21445056       NaN       NaN   \n",
       "1                                                NaN       NaN       NaN   \n",
       "2                                                NaN       NaN       NaN   \n",
       "3  19336573;27759562;16007150;11753428;11900253;2...       NaN       NaN   \n",
       "4  10574788;9219684;20194776;23515407;26619011;29...       NaN       NaN   \n",
       "\n",
       "  LEVEL_Dx3 HIGHEST_DX_LEVEL                             LEVEL_Px1  LEVEL_Px2  \\\n",
       "0       NaN              NaN  AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF        NaN   \n",
       "1       NaN              NaN                                   NaN        NaN   \n",
       "2       NaN              NaN                                   NaN        NaN   \n",
       "3       NaN              NaN  AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF        NaN   \n",
       "4       NaN              NaN                                   NaN        NaN   \n",
       "\n",
       "  LEVEL_Px3 HIGHEST_PX_LEVEL  \n",
       "0       MCL        LEVEL_Px1  \n",
       "1       NaN              NaN  \n",
       "2       NaN              NaN  \n",
       "3       MCL        LEVEL_Px1  \n",
       "4       NaN              NaN  \n",
       "\n",
       "[5 rows x 158 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "flt = pd.read_table(f\"{project}/results/cohort_filtered.maf\") # here, the project directory has been soft-linked to this directory\n",
    "flt.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f46ce691-884b-4869-a7ce-20d6228892c6",
   "metadata": {},
   "source": [
    "#### - MSKCC + Uniprot isoforms "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "21476966-2e36-42fb-979e-c19e1c9a4d2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#enst_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>refseq_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000318560</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>NM_005157.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000321945</td>\n",
       "      <td>ABRAXAS1</td>\n",
       "      <td>NM_139076.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000434821</td>\n",
       "      <td>ACVR1</td>\n",
       "      <td>NM_001111067.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000373204</td>\n",
       "      <td>AGO1</td>\n",
       "      <td>NM_012199.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000220592</td>\n",
       "      <td>AGO2</td>\n",
       "      <td>NM_012154.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          #enst_id gene_name       refseq_id\n",
       "0  ENST00000318560      ABL1     NM_005157.4\n",
       "1  ENST00000321945  ABRAXAS1     NM_139076.2\n",
       "2  ENST00000434821     ACVR1  NM_001111067.2\n",
       "3  ENST00000373204      AGO1     NM_012199.2\n",
       "4  ENST00000220592      AGO2     NM_012154.3"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoform_mskcc = pd.read_table(\"isoform_overrides_at_mskcc.txt\")\n",
    "isoform_mskcc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7defc764-4c4d-4b17-aa00-0e76e939049c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#enst_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>refseq_id</th>\n",
       "      <th>ccds_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000263100</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>NM_130786.3</td>\n",
       "      <td>CCDS12976.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000373993</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>NM_138932.2</td>\n",
       "      <td>CCDS7242.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000318602</td>\n",
       "      <td>A2M</td>\n",
       "      <td>NM_000014.4</td>\n",
       "      <td>CCDS44827.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000299698</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS8596.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000249005</td>\n",
       "      <td>A4GALT</td>\n",
       "      <td>NM_017436.4</td>\n",
       "      <td>CCDS14041.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          #enst_id gene_name    refseq_id      ccds_id\n",
       "0  ENST00000263100      A1BG  NM_130786.3  CCDS12976.1\n",
       "1  ENST00000373993      A1CF  NM_138932.2   CCDS7242.1\n",
       "2  ENST00000318602       A2M  NM_000014.4  CCDS44827.1\n",
       "3  ENST00000299698     A2ML1          NaN   CCDS8596.2\n",
       "4  ENST00000249005    A4GALT  NM_017436.4  CCDS14041.1"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoform_uniprot = pd.read_table(\"isoform_overrides_uniprot.txt\")\n",
    "isoform_uniprot.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe6f78ce-43ab-494b-921e-54305c98938f",
   "metadata": {},
   "source": [
    "#### - Compare Isabl generate_maf result isoforms with MSKCC isoforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ed604c52-2cbe-4eaa-a57a-7246a43b1ccf",
   "metadata": {},
   "outputs": [],
   "source": [
    "mskcc_set = set(isoform_mskcc['#enst_id']) # Isoforms in MSKCC\n",
    "uniprot_set = set(isoform_uniprot['#enst_id']) # Isoforms in UniProt\n",
    "both_set = mskcc_set | uniprot_set # Union (~= both)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c7dfc2d8-3dab-44c6-8b58-42e823503cfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_all_effects(row, both_set):\n",
    "    \"\"\"Checks if there is another isoform name within all_effects column\n",
    "    that is included in both sets (MSKCC + UniProt)\"\"\"\n",
    "    assert hasattr(row, \"all_effects\"), f\"ERROR: row does not have all_effects:\\n{row}\"\n",
    "    isoforms = row.all_effects.split(';')\n",
    "    for isoform in isoforms:\n",
    "        if len(isoform) < 1: continue # skip empty values\n",
    "        if isoform.count(',') >= 4:\n",
    "            features = isoform.split(',', 4)\n",
    "            gene_sym, effect, protein_change, enst_id, nm_id = features\n",
    "        else:\n",
    "            continue\n",
    "        if enst_id in both_set:\n",
    "            return (enst_id, protein_change, effect)\n",
    "        else:\n",
    "            continue\n",
    "    return (\"n/a\", \"n/a\", \"n/a\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a4a6228f-9d1c-430e-8d63-d3d37d346a49",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "included: 10833 (spared: 2147), excluded: 650\n",
      "CPU times: user 689 ms, sys: 17.4 ms, total: 707 ms\n",
      "Wall time: 726 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "def get_excluded_genes(df):\n",
    "    \"\"\"Checks is a representative VEP isoform is in both_set, but if not, \n",
    "    checks again if another isoform from the all_effects column is in both_set,\n",
    "    and if there is such isoform, labels the row 'spared', and if not,\n",
    "    labels the row 'excluded', and returns a list of df indices that are excluded\"\"\"\n",
    "    included = spared = excluded = 0\n",
    "    genes_spared = set()\n",
    "    genes_excluded = set()\n",
    "    idx_excluded = list()\n",
    "    excluded_df = pd.DataFrame(columns = df.columns)\n",
    "    for ix, row in enumerate(df.iterrows()):\n",
    "        row = row[1] # ignore index column\n",
    "        if row.Transcript_ID in both_set: # if ENST transcript id in MSKCC+UniProt database\n",
    "            included += 1\n",
    "        else:\n",
    "            enst_id, protein_change, effect = parse_all_effects(row, both_set) #\n",
    "            if enst_id == 'n/a':\n",
    "                excluded += 1\n",
    "                genes_excluded.add(row.Transcript_ID)\n",
    "                idx_excluded.append(ix)\n",
    "            else:\n",
    "                included += 1\n",
    "                spared += 1\n",
    "                genes_spared.add(enst_id)\n",
    "\n",
    "    print(f'included: {included} (spared: {spared}), excluded: {excluded}') # debug\n",
    "    return idx_excluded\n",
    "\n",
    "#raw_idx_excluded = get_excluded_genes(raw)\n",
    "flt_idx_excluded = get_excluded_genes(flt)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18764de5-7a89-42f3-9899-d7a91031d6bf",
   "metadata": {},
   "source": [
    "##### - Ratio of included (in MSKCC/UniProt), not included but spared, excluded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "70c2e48d-5993-4f41-b530-21eec64beef5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "included (before sparing): 0.76, \n",
      "spared: 0.19,\n",
      "included + spared: 0.94,\n",
      "excluded: 0.06\n"
     ]
    }
   ],
   "source": [
    "ratio_included = (10833-2147)/flt.shape[0]\n",
    "ratio_spared = 2147/flt.shape[0]\n",
    "ratio_spared_plus_included = 10833/flt.shape[0]\n",
    "ratio_excluded = 650/flt.shape[0]\n",
    "\n",
    "print(f\"\"\"included (before sparing): {ratio_included:.2f}, \n",
    "spared: {ratio_spared:.2f},\n",
    "included + spared: {ratio_spared_plus_included:.2f},\n",
    "excluded: {ratio_excluded:.2f}\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32aec50a-630f-456e-a1f0-43828bd416e1",
   "metadata": {},
   "source": [
    "- You can see that 76% isoforms from cohort_filtered maf is not in (MSKCC or UniProt) isoforms\n",
    "- Still, 19% of them still have matching isoforms in their 'all_effects' column\n",
    "- But 6% of them cannot be matched even trying to find a match in the 'all_effects' column"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d6f4baf-d879-4363-8cdf-e9ce8a9fb57e",
   "metadata": {},
   "source": [
    "##### - Counts of Variant_Classification (just to see what types of variants we're not matching/excluding, and matching/including)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "c9d5f815-4e5a-4a76-9e9e-002920f1c2d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Splice_Region      403\n",
      "Splice_Site        245\n",
      "Frame_Shift_Del      2\n",
      "Name: Variant_Classification, dtype: int64\n",
      "Likely Loss-of-function    650\n",
      "Name: MUTATION_EFFECT, dtype: int64\n",
      "Likely Oncogenic    650\n",
      "Name: ONCOGENIC, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# isoforms not-included in our databases\n",
    "print(flt.iloc[flt_idx_excluded].Variant_Classification.value_counts()) \n",
    "print(flt.iloc[flt_idx_excluded].MUTATION_EFFECT.value_counts())\n",
    "print(flt.iloc[flt_idx_excluded].ONCOGENIC.value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "58bac5b7-dd3b-487f-9c46-c8047d6ff20c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Splice_Region             8493\n",
      "Frame_Shift_Ins            544\n",
      "Nonsense_Mutation          424\n",
      "Nonstop_Mutation           364\n",
      "Missense_Mutation          340\n",
      "Splice_Site                261\n",
      "Frame_Shift_Del            222\n",
      "In_Frame_Ins               151\n",
      "Translation_Start_Site      23\n",
      "In_Frame_Del                11\n",
      "Name: Variant_Classification, dtype: int64\n",
      "Likely Loss-of-function    10371\n",
      "Unknown                      379\n",
      "Loss-of-function              44\n",
      "Likely Gain-of-function       18\n",
      "Inconclusive                  12\n",
      "Gain-of-function               9\n",
      "Name: MUTATION_EFFECT, dtype: int64\n",
      "Likely Oncogenic       10411\n",
      "Predicted Oncogenic      229\n",
      "Oncogenic                 31\n",
      "Inconclusive              12\n",
      "Name: ONCOGENIC, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# included (including spared) in our databases\n",
    "print(flt.drop(flt_idx_excluded).Variant_Classification.value_counts())\n",
    "print(flt.drop(flt_idx_excluded).MUTATION_EFFECT.value_counts())\n",
    "print(flt.drop(flt_idx_excluded).ONCOGENIC.value_counts())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4d71ff00-8a35-4f3c-aa1f-15f4526ef4ca",
   "metadata": {},
   "source": [
    "##### Save as CSVs: genes not in OncoKB, and genes in OncoKB - just so for whatever twisted reasons I could access them later"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "59a121a2-029e-46a2-8a69-360ef0825e20",
   "metadata": {},
   "outputs": [],
   "source": [
    "flt.iloc[flt_idx_excluded].to_csv(f\"{project}.cohort_filtered.genes_not_in_oncokb.maf.csv\")\n",
    "flt.drop(flt_idx_excluded).to_csv(f\"{project}.cohort_filtered.genes_in_oncokb.maf.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e602ebba-e4af-4a30-a17d-fab5ec7c46d3",
   "metadata": {},
   "source": [
    "###"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f712b0a-6343-4916-87bb-3a5159a7c2b8",
   "metadata": {},
   "source": [
    "### Topic 2. Annotation differences between: protein-queried results, and genomic position (default)-queried results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a5490e8-b6b1-4fdb-8172-64c9b150a827",
   "metadata": {},
   "source": [
    "While doing this isoforms study I found out that `MafAnnotator.py` results differ (even with the same input) when queried with genomic positions (g-queries) or protein change (p-queries).\n",
    "<br>Actually\n",
    "- p-queries fail to annotate some missense mutations and in-frame deletions, while\n",
    "- q-queries usually fail to annotate some splice site mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b059d5c0-81fd-4ee4-9e7b-818d582cc2e4",
   "metadata": {},
   "source": [
    "##### MafAnnotator output for genomic queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "672dbb12-a9ae-4dce-8470-f3f3c019b0bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_29513/1478954808.py:1: DtypeWarning: Columns (1,2,5,6,8,11,14,17,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,39,40,41,42,43,44,45,46,47,57,58,59,60,61,62,65,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,98,99,100,101,102,103,104,105,106,107,110,111,112,113,114,115,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,140,142,143,144,145,146,147,156,161,163,164,165,167,168,171,172,173,174,175,176,177,178,179,180,181,182) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  g_annot = pd.read_table(\"./query_comparison/merged_filtered.g_annotated.maf\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genomic_vcf37_source</th>\n",
       "      <th>chr_hg19</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>ref_hg19</th>\n",
       "      <th>alt_hg19</th>\n",
       "      <th>clinical_significance_brcaexchange</th>\n",
       "      <th>brcaexchange</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>...</th>\n",
       "      <th>LEVEL_Dx1.1</th>\n",
       "      <th>LEVEL_Dx2.1</th>\n",
       "      <th>LEVEL_Dx3.1</th>\n",
       "      <th>HIGHEST_DX_LEVEL.1</th>\n",
       "      <th>DX_CITATIONS</th>\n",
       "      <th>LEVEL_Px1.1</th>\n",
       "      <th>LEVEL_Px2.1</th>\n",
       "      <th>LEVEL_Px3.1</th>\n",
       "      <th>HIGHEST_PX_LEVEL.1</th>\n",
       "      <th>PX_CITATIONS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7577094</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCL</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "      <td>25412851;25860933;22186996;18596741;25412846;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7578190</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>56477631</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERBB3</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7574003</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCL</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "      <td>25412851;25860933;22186996;18596741;25412846;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>115256530</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CMML,JMML,MDS</td>\n",
       "      <td>ETPLL</td>\n",
       "      <td>LEVEL_Dx2</td>\n",
       "      <td>26341525;23690417;Patel, B. et al., Abstract# ...</td>\n",
       "      <td>MDS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "      <td>24881041;21714648;8329714;24220272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 183 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  genomic_vcf37_source chr_hg19   pos_hg19 ref_hg19 alt_hg19  \\\n",
       "0                  NaN       17    7577094        G        A   \n",
       "1                  NaN       17    7578190        T        C   \n",
       "2                  NaN       12   56477631        T        A   \n",
       "3                  NaN       17    7574003        G        A   \n",
       "4                  NaN        1  115256530        G        T   \n",
       "\n",
       "  clinical_significance_brcaexchange brcaexchange Hugo_Symbol Entrez_Gene_Id  \\\n",
       "0                                NaN          NaN        TP53              0   \n",
       "1                                NaN          NaN        TP53              0   \n",
       "2                                NaN          NaN       ERBB3              0   \n",
       "3                                NaN          NaN        TP53              0   \n",
       "4                                NaN          NaN        NRAS              0   \n",
       "\n",
       "  Center  ... LEVEL_Dx1.1    LEVEL_Dx2.1 LEVEL_Dx3.1 HIGHEST_DX_LEVEL.1  \\\n",
       "0      .  ...         NaN            NaN         NaN                NaN   \n",
       "1      .  ...         NaN            NaN         NaN                NaN   \n",
       "2      .  ...         NaN            NaN         NaN                NaN   \n",
       "3      .  ...         NaN            NaN         NaN                NaN   \n",
       "4      .  ...         NaN  CMML,JMML,MDS       ETPLL          LEVEL_Dx2   \n",
       "\n",
       "                                        DX_CITATIONS  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4  26341525;23690417;Patel, B. et al., Abstract# ...   \n",
       "\n",
       "                            LEVEL_Px1.1 LEVEL_Px2.1 LEVEL_Px3.1  \\\n",
       "0  AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF         NaN         MCL   \n",
       "1                                   NaN         NaN         NaN   \n",
       "2                                   NaN         NaN         NaN   \n",
       "3  AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF         NaN         MCL   \n",
       "4                                   MDS         NaN         NaN   \n",
       "\n",
       "  HIGHEST_PX_LEVEL.1                                       PX_CITATIONS  \n",
       "0          LEVEL_Px1  25412851;25860933;22186996;18596741;25412846;2...  \n",
       "1                NaN                                                NaN  \n",
       "2                NaN                                                NaN  \n",
       "3          LEVEL_Px1  25412851;25860933;22186996;18596741;25412846;2...  \n",
       "4          LEVEL_Px1                 24881041;21714648;8329714;24220272  \n",
       "\n",
       "[5 rows x 183 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g_annot = pd.read_table(\"./query_comparison/merged_filtered.g_annotated.maf\")\n",
    "g_annot.to_csv(\"./query_comparison/merged_filtered.g_annotated.maf.csv\")\n",
    "g_annot.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ba38ceb-80a9-43cd-a7ca-3ff9a90d545c",
   "metadata": {},
   "source": [
    "##### MafAnnotator output for protein queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "81edbd1d-a44f-4ff5-b80d-5a1027ecb44a",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_29513/3966712830.py:1: DtypeWarning: Columns (1,2,5,6,8,11,14,17,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,39,40,41,42,43,44,45,46,47,57,58,59,60,61,62,65,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,98,99,100,101,102,103,104,105,106,107,110,111,112,113,114,115,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,140,142,143,144,145,146,147,156,161,163,164,165,167,168,171,172,173,174,175,176,177,178,179,180,181,182) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  p_annot = pd.read_table(\"./query_comparison/merged_filtered.p_annotated.maf\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genomic_vcf37_source</th>\n",
       "      <th>chr_hg19</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>ref_hg19</th>\n",
       "      <th>alt_hg19</th>\n",
       "      <th>clinical_significance_brcaexchange</th>\n",
       "      <th>brcaexchange</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>...</th>\n",
       "      <th>LEVEL_Dx1.1</th>\n",
       "      <th>LEVEL_Dx2.1</th>\n",
       "      <th>LEVEL_Dx3.1</th>\n",
       "      <th>HIGHEST_DX_LEVEL.1</th>\n",
       "      <th>DX_CITATIONS</th>\n",
       "      <th>LEVEL_Px1.1</th>\n",
       "      <th>LEVEL_Px2.1</th>\n",
       "      <th>LEVEL_Px3.1</th>\n",
       "      <th>HIGHEST_PX_LEVEL.1</th>\n",
       "      <th>PX_CITATIONS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7577094</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCL</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "      <td>25412851;25860933;22186996;18596741;25412846;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7578190</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>56477631</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ERBB3</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>7574003</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TP53</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MCL</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "      <td>25412851;25860933;22186996;18596741;25412846;2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>115256530</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CMML,JMML,MDS</td>\n",
       "      <td>ETPLL</td>\n",
       "      <td>LEVEL_Dx2</td>\n",
       "      <td>26341525;23690417;Patel, B. et al., Abstract# ...</td>\n",
       "      <td>MDS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LEVEL_Px1</td>\n",
       "      <td>24881041;21714648;8329714;24220272</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 183 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  genomic_vcf37_source chr_hg19   pos_hg19 ref_hg19 alt_hg19  \\\n",
       "0                  NaN       17    7577094        G        A   \n",
       "1                  NaN       17    7578190        T        C   \n",
       "2                  NaN       12   56477631        T        A   \n",
       "3                  NaN       17    7574003        G        A   \n",
       "4                  NaN        1  115256530        G        T   \n",
       "\n",
       "  clinical_significance_brcaexchange brcaexchange Hugo_Symbol Entrez_Gene_Id  \\\n",
       "0                                NaN          NaN        TP53              0   \n",
       "1                                NaN          NaN        TP53              0   \n",
       "2                                NaN          NaN       ERBB3              0   \n",
       "3                                NaN          NaN        TP53              0   \n",
       "4                                NaN          NaN        NRAS              0   \n",
       "\n",
       "  Center  ... LEVEL_Dx1.1    LEVEL_Dx2.1 LEVEL_Dx3.1 HIGHEST_DX_LEVEL.1  \\\n",
       "0      .  ...         NaN            NaN         NaN                NaN   \n",
       "1      .  ...         NaN            NaN         NaN                NaN   \n",
       "2      .  ...         NaN            NaN         NaN                NaN   \n",
       "3      .  ...         NaN            NaN         NaN                NaN   \n",
       "4      .  ...         NaN  CMML,JMML,MDS       ETPLL          LEVEL_Dx2   \n",
       "\n",
       "                                        DX_CITATIONS  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4  26341525;23690417;Patel, B. et al., Abstract# ...   \n",
       "\n",
       "                            LEVEL_Px1.1 LEVEL_Px2.1 LEVEL_Px3.1  \\\n",
       "0  AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF         NaN         MCL   \n",
       "1                                   NaN         NaN         NaN   \n",
       "2                                   NaN         NaN         NaN   \n",
       "3  AMLMRC,AML,TMN,CLLSLL,MDS,ET,MPN,PMF         NaN         MCL   \n",
       "4                                   MDS         NaN         NaN   \n",
       "\n",
       "  HIGHEST_PX_LEVEL.1                                       PX_CITATIONS  \n",
       "0          LEVEL_Px1  25412851;25860933;22186996;18596741;25412846;2...  \n",
       "1                NaN                                                NaN  \n",
       "2                NaN                                                NaN  \n",
       "3          LEVEL_Px1  25412851;25860933;22186996;18596741;25412846;2...  \n",
       "4          LEVEL_Px1                 24881041;21714648;8329714;24220272  \n",
       "\n",
       "[5 rows x 183 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p_annot = pd.read_table(\"./query_comparison/merged_filtered.p_annotated.maf\")\n",
    "p_annot.to_csv(\"./query_comparison/merged_filtered.p_annotated.maf.csv\")\n",
    "p_annot.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ddcc106-e26d-4369-9653-cfa8d67e1f2a",
   "metadata": {},
   "source": [
    "My apologies! `'VARIANT_IN_ONCOKB'` column was made by MafAnnotator when the ran with the default options (genomic queryig),\n",
    "<br>while `'VARIANT_IN_ONKOKB.1'` column was added by MafAnnotator when I used the MafAnnotator output again as input, using the protein querying option"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "ca6d90da-fe8b-4d92-a016-6858dd27e3cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "missense_variant    17\n",
       "inframe_deletion    13\n",
       "Name: Consequence, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Variants with: protein-queried annotations in OncoKB & genomic-queried annotations NOT in OncoKB\n",
    "pO_gX = p_annot[(p_annot['VARIANT_IN_ONCOKB'] != True) & (p_annot['VARIANT_IN_ONCOKB.1'] == True)]\n",
    "pO_gX.to_csv(\"merged_filtered.g-X_p-O.maf.csv\")\n",
    "pO_gX['Consequence'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "2d2a5d7b-9959-4d2c-9bea-d373c3418137",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "missense_variant                                                  13\n",
       "splice_donor_variant                                              13\n",
       "splice_acceptor_variant                                            9\n",
       "splice_donor_variant,frameshift_variant                            9\n",
       "frameshift_variant                                                 9\n",
       "splice_acceptor_variant,coding_sequence_variant,intron_variant     5\n",
       "stop_gained                                                        4\n",
       "splice_acceptor_variant,coding_sequence_variant                    1\n",
       "splice_donor_variant,missense_variant                              1\n",
       "splice_donor_variant,intron_variant                                1\n",
       "Name: Consequence, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Variants with: protein-queried annotations NOT in OncoKB & genomic-queried annotations in OncoKB\n",
    "pX_gO = p_annot[(p_annot['VARIANT_IN_ONCOKB'] == True) & (p_annot['VARIANT_IN_ONCOKB.1'] != True)]\n",
    "pX_gO.to_csv(\"merged_filtered.g-O_p-X.maf.csv\")\n",
    "pX_gO['Consequence'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5e7bc99-9864-4fbc-95aa-502e70070675",
   "metadata": {},
   "source": [
    "You can see that MafAnnotator results differ by the querying option of MafAnnotator.py\n",
    "<br>Interestingly, the Mutation consequence differ by the querying option itself!"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
